### PR TITLE
Prevent creation of multiple default storage classes

### LINKED ISSUE #48416: Two or more default StorageClasses can be created
/kind bug

DefaultStorageClass admission plugin expects that there is only one default StorageClass defined and it refuses to work when there are more of them. The admission plugin should be extended to check newly created/updated StorageClassses and forbid creation of two default ones.

cc @kubernetes/sig-storage-bugs 
/assign

### PR DESCRIPTION
**Pull request by jsafrane**
DefaultStorageClass admission plugin should not allow creation of more than one default storage class. Users are complaining that they should be notified when creating the second default class and not when creating a PVC.

Fixes #48416

@thockin, I remember we talked about something similar when designing StorageClasses, but I don't remember exactly why we decided to allow the users to create two or more default storage classes. I think the argument was that there may be a different defaulting policy in a different admission plugin. This PR does change that, a different admission plugin may implement whatever it wants, just our DefaultStorageClass won't work with two default storage classes and thus it should refuse creation / modification of the second default StorageClass instance.

```release-note
NONE
```

### CODE REVIEW

File: plugin/pkg/admission/storageclass/setdefault/admission.go [hunk 1]
```diff
@@ -77,17 +77,61 @@ func (a *claimDefaulterPlugin) Validate() error {
...
+	if !ok {
+		return nil
+	}
+
+	if !storageutil.IsDefaultAnnotation(class.ObjectMeta) {
+		// this class is not default
+		return nil
+	}
+
+	def, err := getDefaultClass(c.lister)
```
Comment (gnufied) [inline plugin/pkg/admission/storageclass/setdefault/admission.go]: Can we change this variable name to something more readable. like `defaultClass` ?

**Comment by thockin** [comment]
Idea is sound.  This is not a code review.

This needs an issue opened and a `fixes #12345` in the FIRST COMMENT before it can be approved.

**Comment by jsafrane** [comment]
`fixes #48416` added.

File: plugin/pkg/admission/storageclass/setdefault/admission.go (hunk 1 above)
Comment (jsafrane) [inline plugin/pkg/admission/storageclass/setdefault/admission.go]: renamed

**Comment by thockin** [comment]
@gnufied or @jingxu97 or @saad-ali  for LGTM

**Comment by thockin** [comment]
ping?

**Comment by jsafrane** [comment]
@thockin, we had a discussion with @liggitt that there may be two or more API servers, each running its own admission controller. Any admission plugin that makes sure only one default StorageClass exists is therefore racy.

See issue #48416 for possible solutions, but that's where we stopped. TBH, I am too lazy to create the first external admission plugin shipped by Kubernetes and solve all deployment issues.

**Pull request closed without merging**